src/main/kotlin/org/kafkalytic/plugin/KafkaUtils.kt

```diff
@@ -223,7 +225,22 @@ fun produceGeneratedMessages(producer: KafkaProducer<ByteArray, ByteArray>, topi
                 }.toString()
             }
 
-            channel.send(producer.send(ProducerRecord<ByteArray, ByteArray>(topic, "message$current".toByteArray(), template.replace(RANDOM_PLACEHOLDER, value).toByteArray())))
+            val listCustomHeader: List<String> = header?.split(";")
+
+            if (listCustomHeader.isNullOrEmpty()) {
```

Comment:
my preference would be instead of code
if (header.isNullorEmpty) {
     send(without header)
} else {
    header = createHeader
    send(with header) 
}
use this approach:
header = createHeader (empty collection is header is null)
send(with header)